### ISSUE TITLE
File name information gets lost

### ISSUE THREAD

**Issue by salomvary** [nodyn/jvm-npm#11]
Resulting in exceptions hard to track down. Like this: `ReferenceError: "foobar" is not defined <function> 3`.

Using Nashorn's [`load()`](https://docs.oracle.com/javase/8/docs/technotes/guides/scripting/nashorn/shell.html#sthref28) instead of `new Function` might be a solution:

```
load({name:"script.js", script:"var x = 1 + 1; x;"})
```

**Comment by jwulf** [comment]
Yeah, this kinda blows. Any way to get better stack traces out of this?
